mongodb/configure.sls

```diff
@@ -0,0 +1,79 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+copy_mongodb_key_file:
+  file.managed:
+    - name: {{ mongodb.cluster_key_file }}
+    - contents: "{{ salt.pillar.get('mongodb:cluster_key') }}"
+    - owner: mongodb
+    - group: mongodb
+    - mode: 0600
+    - require:
+      - pkg: install_packages
+
+stop_mongodb_service:
+  service.dead:
+    - running: True
+    - name: mongodb
+    - enable: True
+    - require:
+      - pkg: install_packages
+
+place_mongodb_config_file:
+  file.managed:
+    - name: /etc/mongod.conf
+    - template: jinja
+    - source: salt://mongodb/templates/mongod.conf.j2
+    - require:
+      - service: stop_mongodb_service
+
+wait_for_mongo:
+  cmd.run:
+    - name: |
+        until [ `netstat -tunlp | grep {{ mongodb.port }} | wc -l` -eq 1 ]
+        do
+          sleep 1
+        done
+    - shell: /bin/bash
+    - timeout: 60
+    - require:
+      - file: place_mongodb_config_file
+      - service: start_mongodb_service
+
+{% if 'mongo_primary' in grains['roles'] %}
+add_admin_user:
+  mongodb_user.present:
+    - name: {{ salt.pillar.get('mongodb:admin_username') }}
+    - passwd: {{ salt.pillar.get('mongodb:admin_password') }}
+    - database: admin
+    - host: localhost
+    - port: {{ mongodb.port }}
+    - require:
+      - file: place_mongodb_config_file
+      - cmd: wait_for_mongo
+
+{% for user in salt.pillar.get('mongodb:users', {}) %}
+add_admin_user:
+  mongodb_user.present:
+    - name: {{ user.user }}
```

Comment:
Maybe change `user.user` to be `user.name`. Your call, but I feel like it's a bit more descriptive.